tony-azkaban/src/main/java/com/linkedin/tony/azkaban/TensorFlowJob.java

```diff
@@ -0,0 +1,195 @@
+   * the status was an error or a timeout
+   */
+  @Override
+  public void run() throws Exception {
+    getLog().info("Hello world from TensorFlow!");
+    setupHadoopOpts(getJobProps());
+    if (shouldProxy && obtainTokens) {
+      getLog().info("Need to proxy. Getting tokens.");
+      Props props = new Props();
+      props.putAll(getJobProps());
+      props.putAll(getSysProps());
+
+      tokenFile = HadoopJobUtils.getHadoopTokens(hadoopSecurityManager, props, getLog());
+      getJobProps().put("env." + HADOOP_TOKEN_FILE_LOCATION, tokenFile.getAbsolutePath());
+    }
+    super.run();
+  }
+
+  private void setupHadoopOpts(Props props) {
+    if (props.containsKey(HADOOP_GLOBAL_OPTS)) {
+      String hadoopGlobalOps = props.getString(HADOOP_GLOBAL_OPTS);
+      if (props.containsKey(HADOOP_OPTS)) {
+        String hadoopOps = props.getString(HADOOP_OPTS);
+        props.put(HADOOP_OPTS, String.format("%s %s", hadoopOps, hadoopGlobalOps));
+      } else {
+        props.put(HADOOP_OPTS, hadoopGlobalOps);
+      }
+    }
+  }
+
+  @Override
+  public String getJavaClass() {
+    return "com.linkedin.tony.TonyClient";
+  }
+
+  @Override
+  protected String getJVMArguments() {
+    String args = super.getJVMArguments();
+
+    String typeUserGlobalJVMArgs = getJobProps().getString(HadoopJobUtils.JOBTYPE_GLOBAL_JVM_ARGS, null);
+    if (typeUserGlobalJVMArgs != null) {
+      args += " " + typeUserGlobalJVMArgs;
+    }
+    String typeSysGlobalJVMArgs = getSysProps().getString(HadoopJobUtils.JOBTYPE_GLOBAL_JVM_ARGS, null);
+    if (typeSysGlobalJVMArgs != null) {
+      args += " " + typeSysGlobalJVMArgs;
+    }
+    String typeUserJVMArgs = getJobProps().getString(HadoopJobUtils.JOBTYPE_JVM_ARGS, null);
+    if (typeUserJVMArgs != null) {
+      args += " " + typeUserJVMArgs;
+    }
+    String typeSysJVMArgs = getSysProps().getString(HadoopJobUtils.JOBTYPE_JVM_ARGS, null);
+    if (typeSysJVMArgs != null) {
+      args += " " + typeSysJVMArgs;
+    }
+
+    String typeUserJVMArgs2 = getJobProps().getString(HadoopJobUtils.JVM_ARGS, null);
+    if (typeUserJVMArgs != null) {
+      args += " " + typeUserJVMArgs2;
+    }
+    String typeSysJVMArgs2 = getSysProps().getString(HadoopJobUtils.JVM_ARGS, null);
+    if (typeSysJVMArgs != null) {
+      args += " " + typeSysJVMArgs2;
+    }
+    return args;
+  }
+
+  @Override
+  protected String getMainArguments() {
+    String args = super.getMainArguments();
+    info("All job props: " + getJobProps());
+    info("All sys props: " + getSysProps());
+    String srcDir = getJobProps().getString(TensorFlowJobArg.SRC_DIR.azPropName, "src");
+    args += " " + TensorFlowJobArg.SRC_DIR.tfParamName + " " + srcDir;
+
+    String hdfsClasspath = getJobProps().getString(TensorFlowJobArg.HDFS_CLASSPATH.azPropName, null);
+    if (hdfsClasspath != null) {
+      args += " " + TensorFlowJobArg.HDFS_CLASSPATH.tfParamName + " " + hdfsClasspath;
```

Comment:
HDFS_CLASSPATH should be deprecated

Reply:
I don't think we should deprecate this. In general users should run with hdfs_classpath. I don't think it's a good practice to use the fatjar approach, even if it's easiest.

Reply:
What is the disadvantage of using a farjar here? it only makes user's life easier. 
- it avoids the jar conflicts problem
- it prevents the problem is a user forgot to run the cachedLib task

Reply:
What jar conflict problem are you referring to? Using fatjar makes dependency conflicts worse. e.g if they are running one version of Hadoop which has a dependency with version X, and TonY is built against another version which pulls in that dependency with version Y, there will be a mismatch